src/compat.h

```diff
@@ -26,58 +38,360 @@
+ * fixed tupleDesc when making a TupleTableSlot, which can be useful during
+ * execution for JITted operations. (See:
+ * https://github.com/postgres/postgres/commit/ad7dbee368a7cd9e595d2a957be784326b08c943).
+ * We adopt the PG11 conventions so that we can take advantage of JITing more easily in the future.
+ */
+#if PG96 || PG10
+static inline  TupleTableSlot * ExecInitExtraTupleSlotCompat(EState *estate, TupleDesc tupdesc) {
+	TupleTableSlot *myslot = ExecInitExtraTupleSlot(estate);
+
+	if(tupdesc != NULL)
+		ExecSetSlotDescriptor(myslot, tupdesc);
+
+	return myslot;
+}
+
+static inline TupleTableSlot * MakeTupleTableSlotCompat(TupleDesc tupdesc) {
+	TupleTableSlot *myslot = MakeTupleTableSlot();
+
+	if(tupdesc != NULL)
+		ExecSetSlotDescriptor(myslot, tupdesc);
+
+	return myslot;
+}
+#else
+#define ExecInitExtraTupleSlotCompat ExecInitExtraTupleSlot
+#define MakeTupleTableSlotCompat MakeTupleTableSlot
+#endif
+
+/*
+ * get_attname
+ *
+ * PG11 introduced a missing_ok boolean to the function signature. Given that
+ * that seems like a useful check in many cases, we are going with forwards
+ * compatbility here and have a small static inline function to replicate the
+ * behavior on older versions.
+ */
+#if PG96 || PG10
+static inline char * get_attname_compat(Oid relid, AttrNumber attnum, bool missing_ok){
+	char *  name = get_attname(relid, attnum);
+
+	if (!missing_ok && name == NULL)
+			elog(ERROR, "cache lookup failed for attribute %d of relation %u",
+				attnum, relid);
+	return name;
+}
+#else
+#define get_attname_compat get_attname
+#endif
+
+/* get_projection_info_slot */
+#if PG96
+#define get_projection_info_slot_compat(pinfo) \
+	(pinfo->pi_slot)
+#else
+#define get_projection_info_slot_compat(pinfo) \
+	(pinfo->pi_state.resultslot)
+#endif
+
+/*
+ * heap_attisnull
+ *
+ * PG11 modified how heap tuples are accessed, especially for null values, to
+ * allow for fast alters when tables have not-null default (with non-volatile
+ * functions).
+ * See: https://github.com/postgres/postgres/commit/16828d5c0273b4fe5f10f42588005f16b415b2d8
+ * It now requires the passing in of  a TupleDesc in a number of places where it
+ * didn't before. We adopt the PG11 convention of passing in a TupleDesc (which
+ * can be null for pg_catalog tables, but must be provided otherwise), and
+ * simply omit in earlier versions.
+ */
+#if PG96 || PG10
+#define heap_attisnull_compat(tup, attnum, tupledesc) \
+	heap_attisnull(tup, attnum)
+#else
+#define heap_attisnull_compat heap_attisnull
+#endif
+
+/*
+ * index_create
+ *
+ * PG11 replaced several boolean arguments to index_create with a single bitmask
+ * flag. Since that is generally cleaner, we adopt that convention. Several new
+ * arguments were also introduced, so we introduce a compat that takes the best
+ * of both worlds. (See:
+ * https://github.com/postgres/postgres/commit/a61f5ab986386628cf20b33971364475ce452412)
+
+ * add note about parents etc. for future
```

Comment:
What does this line refer to?

Reply:
Seems like it was a note for myself, and now I don't remember at all, ugh. Will have to look back at the other PR and see if I can remind myself.